File: pset6/caesar.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("You should provide cmd line arguments!")
        exit(1)
    key = int(sys.argv[1])
    i = 0
    ciphertext = 0
    if(key <0):
        print("please enter a valid key")
    else:
        print("plaintext: ")
        name = cs50.get_string()
        print("ciphertext: ")
        for i in range (i,len(name),+1):
            if(name[i].isalpha()):
                if(name[i].isupper()):
                    ciphertext = ((ord(name[i]) - 65 + key) % 26) + 65
                    print(chr(ciphertext),end = "")
                elif(name[i].islower()):
                    ciphertext  = ((ord(name[i]) - 97 + key) % 26) + 97
                    print(chr(ciphertext),end = "")
            else:
                print(name[i])
if __name__ == "__main__":
    main()

File: pset2/caesar.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <cs50.h>
int main(int argc, string argv[]) {
  if(argc!=2) {
      printf("Provide Exactly one argument");
      return 1;
  } 
    int key = atoi(argv[1]);
    if(key <0) {
        printf("please enter a valid key");
        return 1;
    }
    else {
        printf("plaintext: ");
        string name = get_string();
        printf("ciphertext: ");
        for(int i = 0;i<strlen(name);i++) {
            if(isalpha(name[i])) {
            if(isupper(name[i])) {
                name[i] = (((name[i]- 65) + key) % 26) + 65;
                printf("%c",name[i]);
            }
            else if(islower(name[i])) {
                name[i] = (((name[i]- 97) + key) % 26) + 97;
                printf("%c",name[i]);
            }
            else {
                printf("%c",name[i]);
            }
        }
        else
        printf("%c",name[i]);
        
        }
        printf("\n");
    }
    return 0;
}